pkg/remoteclient/fake.go

```diff
@@ -0,0 +1,142 @@
+package remoteclient
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	configv1 "github.com/openshift/api/config/v1"
+	openshiftapiv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	autoscalingv1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1"
+	autoscalingv1beta1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+// fakeBuilder builds fake clients for fake clusters. Used to simulate communication with a cluster
+// that doesn't actually exist in scale testing.
+type fakeBuilder struct {
+	c              client.Client
+	cd             *hivev1.ClusterDeployment
+	controllerName hivev1.ControllerName
+	urlToUse       int
+}
+
+// Build returns a fake controller-runtime test client populated with the resources we expect to query for a
+// fake cluster.
+func (b *fakeBuilder) Build() (client.Client, error) {
+	scheme, err := machineapi.SchemeBuilder.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	autoscalingv1.SchemeBuilder.AddToScheme(scheme)
+	autoscalingv1beta1.SchemeBuilder.AddToScheme(scheme)
+
+	if err := openshiftapiv1.Install(scheme); err != nil {
+		return nil, err
+	}
+
+	if err := routev1.Install(scheme); err != nil {
+		return nil, err
+	}
```

Comment:
some AddToScheme has error handling, while the previous one's don't any reason why in particular?

Reply:
Old code coming over from the regular builder creation, I'll see if I can clean them up.

Reply:
New commit added for this, a couple don't appear to have an Install function only AddToScheme. Added error handling for all.